cypress/e2e/swap/uniswapx.test.ts

```diff
@@ -1,32 +1,55 @@
-import { DAI, USDC_MAINNET } from '../../../src/constants/tokens'
+import { ChainId, CurrencyAmount } from '@uniswap/sdk-core'
+
+import { DAI, nativeOnChain, USDC_MAINNET } from '../../../src/constants/tokens'
 import { getTestSelector } from '../../utils'
 
 const QuoteEndpoint = 'https://api.uniswap.org/v2/quote'
 const QuoteWhereUniswapXIsBetter = 'uniswapx/quote1.json'
+const QuoteWithEthInput = 'uniswapx/quote2.json'
+
+const OrderSubmissionEndpoint = 'https://api.uniswap.org/v2/order'
+const OrderResponse = 'uniswapx/orderResponse.json'
+
+const OrderStatusEndpoint =
+  'https://api.uniswap.org/v2/orders?swapper=0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266&orderHashes=0xa9dd6f05ad6d6c79bee654c31ede4d0d2392862711be0f3bc4a9124af24a6a19'
+const OpenStatusResponse = 'uniswapx/openStatusResponse.json'
+const FilledStatusResponse = 'uniswapx/filledStatusResponse.json'
+const ExpiredStatusResponse = 'uniswapx/expiredStatusResponse.json'
+const TransactionReceipt = 'uniswapx/fillTransactionReceipt.json'
 
-/** Simulates the user opening swap settings and toggling UniswapX.  */
-function toggleUniswapXInSwapSettings() {
-  cy.get(getTestSelector('open-settings-dialog-button')).click()
-  cy.get(getTestSelector('toggle-uniswap-x-button')).click()
-  cy.get(getTestSelector('open-settings-dialog-button')).click()
+const InsufficientFundsStatusResponse = 'uniswapx/insufficientFundsStatusResponse.json'
+
+/** Stubs the provider to return a tx receipt corresponding to the mock filled uniswapx order's txHash */
+function stubSwapTxReceipt() {
+  cy.hardhat().then((hardhat) => {
+    cy.fixture(TransactionReceipt).then((mockTxReceipt) => {
+      const getTransactionReceiptStub = cy.stub(hardhat.provider, 'getTransactionReceipt').log(false)
+      getTransactionReceiptStub.withArgs(mockTxReceipt.transactionHash).resolves(mockTxReceipt)
+      getTransactionReceiptStub.callThrough()
+    })
+  })
 }
 
 describe('UniswapX Toggle', () => {
   beforeEach(() => {
     cy.intercept(QuoteEndpoint, { fixture: QuoteWhereUniswapXIsBetter })
     cy.visit(`/swap/?inputCurrency=${USDC_MAINNET.address}&outputCurrency=${DAI.address}`)
+
+    // Hide banner that partially covers UniswapX mustache
+    cy.get(getTestSelector('uniswap-wallet-banner')).click()
```

Comment:
This should be hidden by setting the localStorage bit that says it's been seen already, not by clicking on it. Otherwise any other changes to it will cause this test to fail.

I don't think the banner is tested in any e2e tests, so it would be fine to make this a global thing in setup.